src/backend/gporca/libgpopt/src/init.cpp

```diff
@@ -41,21 +40,11 @@ static CMemoryPool *mp = NULL;
 void
 gpopt_init()
 {
-	{
-		CAutoMemoryPool amp;
-		mp = amp.Pmp();
+	mp = CMemoryPoolManager::GetMemoryPoolMgr()->CreateMemoryPool();
```

Comment:
Do this change and the same changes below relate to OOM?

Reply:
Yes, this was done to avoid calling the auto memory pool's leaky destructor with an exception.